flax/linen/module.py

```diff
@@ -132,16 +137,29 @@ def _get_suffix_value_pairs(
     tree_or_leaf: Any) -> List[Tuple[str, Type["Module"]]]:
   """Helper for naming pytrees of submodules."""
   dict_or_leaf = serialization.to_state_dict(tree_or_leaf)
-  if dict_or_leaf == {} or not isinstance(dict_or_leaf, dict):
+  if not isinstance(dict_or_leaf, dict) or dict_or_leaf == {}:
```

Comment:
btw: switching the test order here is important since numpy arrays (but not jax arrays!) will throw an exception if compared to {} then cast to boolean - in the numpy case one ends up with an array of `False`s that can't be converted to a bool.